src/Hamiltonians/ExtendedHubbardReal1D.jl

```diff
@@ -64,21 +78,22 @@ function extended_bose_hubbard_interaction(b::SingleComponentFockAddress)
         reg_result += curr.occnum * (curr.occnum - 1)
         prev = curr
     end
-
-    # Handle periodic boundaries
-    last = ifelse(omm[end].mode == num_modes(b), omm[end], zero(eltype(omm)))
-    first = ifelse(omm[1].mode == 1, omm[1], zero(eltype(omm)))
-    ext_result += last.occnum * first.occnum
+    if h.hardwall == false
+        # Handle periodic boundaries
+        last = ifelse(omm[end].mode == num_modes(b), omm[end], zero(eltype(omm)))
+        first = ifelse(omm[1].mode == 1, omm[1], zero(eltype(omm)))
+        ext_result += last.occnum * first.occnum
+    end
 
     return ext_result, reg_result
 end
 
 function diagonal_element(h::ExtendedHubbardReal1D, b::SingleComponentFockAddress)
-    ebhinteraction, bhinteraction = extended_bose_hubbard_interaction(b)
+    ebhinteraction, bhinteraction = extended_hubbard_interaction(h,b)
```

Comment:
```suggestion
    ebhinteraction, bhinteraction = extended_hubbard_interaction(h, b)
```
Similar to comment above. There should be a space after a comma.

Reply:
This was marked resolved, but not actually resolved.